File: atcoder/tyama_atcodertenka1~2013qualbE_golf.go
//usr/bin/env go run $0 $@;exit
package main
import "fmt"
func main(){
	fmt.Println(100)
	fmt.Println("1 0")
	for i:=99;i>1;i-- {fmt.Printf("%d 0\n",i)}
	fmt.Println("0 0")
}

File: atcoder/tyama_atcodertenka1~2013qualbE_golf.hs
#!/usr/bin/env runghc
run n
	| n>1 = do
		putStr (show n)
		putStrLn " 0"
		run (n-1)
	| otherwise = return ()

main = do
	putStrLn "100\n1 0"
	run 99
	putStrLn "0 0"